replica/table.cc

```diff
@@ -3514,8 +3514,9 @@ future<db::replay_position> table::discard_sstables(db_clock::time_point truncat
     // materialized view was created right after truncation started, and it
     // would not have compaction disabled when this function is called on it.
     if (!schema()->is_view()) {
-        auto compaction_disabled = std::ranges::all_of(storage_groups() | std::views::values,
-                                                       std::mem_fn(&storage_group::compaction_disabled));
+        auto compaction_disabled = std::ranges::all_of(storage_groups() | std::views::values, [] (const storage_group_ptr& sgp) {
+            return sgp->async_gate().is_closed() || sgp->compaction_disabled();
+        });
```

Comment:
A comment explaining why we exclude stopped groups would be good. Even in the context of truncate, it would be fine (let's assume truncate doesn't exclude with migration), since a tablet migrated away will have all its data removed anyway on cleanup.